test/unit/org/apache/cassandra/tools/SecondaryIndexSSTableExportTest.java

```diff
@@ -0,0 +1,257 @@
+     
+    @BeforeClass
+    public static void beforeClass()
+    {
+        System.setProperty(org.apache.cassandra.tools.Util.ALLOW_TOOL_REINIT_FOR_TEST, "true"); // Necessary for testing
+    }
+    
+    @AfterClass
+    public static void afterClass()
+    {
+        System.clearProperty(org.apache.cassandra.tools.Util.ALLOW_TOOL_REINIT_FOR_TEST);   
+    }
+    
+    @Test
+    public void testRegularColumnIndex() throws Throwable
+    {
+        String createTable = "CREATE TABLE %s (k int PRIMARY KEY, v int)";
+        String createIndex = "CREATE INDEX ON %s (v)";
+        String insert = "INSERT INTO %s (k, v) VALUES (0, 0)";
+        Pair<String, String> tableIndex = generateSstable(createTable, createIndex, insert);
+        ColumnFamilyStore cfs = getColumnFamilyStore(KEYSPACE, tableIndex.left);
+        assertTrue(cfs.indexManager.hasIndexes());
+        assertNotNull(cfs.indexManager.getIndexByName(tableIndex.right));
+        for(ColumnFamilyStore columnFamilyStore : cfs.indexManager.getAllIndexColumnFamilyStores())
+        {
+            assertTrue(columnFamilyStore.isIndex());
+            for(SSTableReader sst : columnFamilyStore.getLiveSSTables())
+            {
+                String file = sst.getFilename();
+                ToolRunner.ToolResult tool = ToolRunner.invokeClass(SSTableExport.class, file);
+                List<Map<String, Object>> parsed = mapper.readValue(tool.getStdout(), jacksonListOfMapsType);
+                assertNotNull(tool.getStdout(), parsed.get(0).get("partition"));
+                assertNotNull(tool.getStdout(), parsed.get(0).get("rows"));
+            }
+        }
+    }
+
+    @Test
+    public void testPartitionKeyIndex() throws Throwable
+    {
+        String createTable = "CREATE TABLE %s (k int, v int, c text, primary key((k, v)))";
+        String createIndex = "CREATE INDEX ON %s (k)";
+        String insert = "INSERT INTO %s (k, v) VALUES (0, 0)";
+        Pair<String, String> tableIndex = generateSstable(createTable, createIndex, insert);
+        ColumnFamilyStore cfs = getColumnFamilyStore(KEYSPACE, tableIndex.left);
+        assertTrue(cfs.indexManager.hasIndexes());
+        assertNotNull(cfs.indexManager.getIndexByName(tableIndex.right));
+        for(ColumnFamilyStore columnFamilyStore : cfs.indexManager.getAllIndexColumnFamilyStores())
+        {
+            assertTrue(columnFamilyStore.isIndex());
+            for(SSTableReader sst : columnFamilyStore.getLiveSSTables())
+            {
+                String file = sst.getFilename();
+                ToolRunner.ToolResult tool = ToolRunner.invokeClass(SSTableExport.class, file);
+                List<Map<String, Object>> parsed = mapper.readValue(tool.getStdout(), jacksonListOfMapsType);
+                assertNotNull(tool.getStdout(), parsed.get(0).get("partition"));
+                assertNotNull(tool.getStdout(), parsed.get(0).get("rows"));
+            }
+        }
```

Comment:
This block of code is repeated verbatim 8 times in this file. I would encapsulate it into a method so the 8 tests using it can be one-liners. That should save us more that one hundred lines of duplicated code:
```java
@Test
public void testCollectionSetIndex() throws Throwable
{
    testIndex("CREATE TABLE %s (k int , v int, s  text static, c bigint, m map<bigint, text>, l list<text>, st set<int>, primary key((k, v), c))",
              "CREATE INDEX ON %s (st)", 
              "INSERT INTO %s (k, v, s, c, m, l, st) VALUES (0, 0, 's', 10, {100:'v'}, ['l1', 'l2'], {1, 2, 3})");
}

private void testIndex(String createTable, String createIndex, String insert) throws Throwable
{
    Pair<String, String> tableIndex = generateSstable(createTable, createIndex, insert);
    ColumnFamilyStore cfs = getColumnFamilyStore(KEYSPACE, tableIndex.left);
    assertTrue(cfs.indexManager.hasIndexes());
    assertNotNull(cfs.indexManager.getIndexByName(tableIndex.right));

    for (ColumnFamilyStore columnFamilyStore : cfs.indexManager.getAllIndexColumnFamilyStores())
    {
        assertTrue(columnFamilyStore.isIndex());
        for (SSTableReader sst : columnFamilyStore.getLiveSSTables())
        {
            String file = sst.getFilename();
            ToolRunner.ToolResult tool = ToolRunner.invokeClass(SSTableExport.class, file);
            List<Map<String, Object>> parsed = mapper.readValue(tool.getStdout(), jacksonListOfMapsType);
            assertNotNull(tool.getStdout(), parsed.get(0).get("partition"));
            assertNotNull(tool.getStdout(), parsed.get(0).get("rows"));
        }
    }
}
```